File: hw1/pow_full.rb
#! /usr/bin/env ruby
def pow(n, p)
  if(p == 0)
    return 1
  end

  return n * pow(n, p-1)
end

if ARGV.length != 2
  puts "fact.rb usage: [NUMBER] [POWER]" 
  exit
end

puts pow(ARGV[0].to_i, ARGV[1].to_i)

File: hw1/pow_full.py
#! /usr/bin/env python

import sys

def pow(n, p):
  if p == 0:
    return 1
  else:
    return n * pow(n, p-1) 

if len(sys.argv) != 3:
  print("%s usage: [NUMBER] [power]" % sys.argv[0])
  exit()

print(pow(int(sys.argv[1]), int(sys.argv[2])))
